From 2700946d7bdf7a112d79db018b5f8503bd2ffa0d Mon Sep 17 00:00:00 2001
From: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Date: Tue, 21 Jul 2026 01:30:08 +0000
Subject: [PATCH 06473/15333] ASoC: fsl: p1022_ds: add card_to_mdata() macro

p1022_ds.c will be updated when Card capsuling.
To makes its review easy, adds card_to_mdata() and useit to reduce
un-related diff.
No functional change, but is preparation for cleanup driver.

Signed-off-by: Kuninori Morimoto <kuninori.morimoto.gx@renesas.com>
Link: https://patch.msgid.link/875x29jfj4.wl-kuninori.morimoto.gx@renesas.com
Signed-off-by: Mark Brown <broonie@kernel.org>
---
 sound/soc/fsl/p1022_ds.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/sound/soc/fsl/p1022_ds.c b/sound/soc/fsl/p1022_ds.c
index db07817a8024..98389f53246c 100644
--- a/sound/soc/fsl/p1022_ds.c
+++ b/sound/soc/fsl/p1022_ds.c
@@ -73,6 +73,8 @@ struct machine_data {
 	char platform_name[2][DAI_NAME_SIZE]; /* One for each DMA channel */
 };
 
+#define card_to_mdata(_card) container_of(_card, struct machine_data, card)
+
 /**
  * p1022_ds_machine_probe: initialize the board
  *
@@ -82,8 +84,7 @@ struct machine_data {
  */
 static int p1022_ds_machine_probe(struct snd_soc_card *card)
 {
-	struct machine_data *mdata =
-		container_of(card, struct machine_data, card);
+	struct machine_data *mdata = card_to_mdata(card);
 	struct ccsr_guts __iomem *guts;
 
 	guts = ioremap(guts_phys, sizeof(struct ccsr_guts));
@@ -122,8 +123,7 @@ static int p1022_ds_machine_probe(struct snd_soc_card *card)
 static int p1022_ds_startup(struct snd_pcm_substream *substream)
 {
 	struct snd_soc_pcm_runtime *rtd = snd_soc_substream_to_rtd(substream);
-	struct machine_data *mdata =
-		container_of(rtd->card, struct machine_data, card);
+	struct machine_data *mdata = card_to_mdata(rtd->card);
 	struct device *dev = rtd->card->dev;
 	int ret = 0;
 
@@ -156,8 +156,7 @@ static int p1022_ds_startup(struct snd_pcm_substream *substream)
  */
 static int p1022_ds_machine_remove(struct snd_soc_card *card)
 {
-	struct machine_data *mdata =
-		container_of(card, struct machine_data, card);
+	struct machine_data *mdata = card_to_mdata(card);
 	struct ccsr_guts __iomem *guts;
 
 	guts = ioremap(guts_phys, sizeof(struct ccsr_guts));
@@ -399,8 +398,7 @@ static int p1022_ds_probe(struct platform_device *pdev)
 static void p1022_ds_remove(struct platform_device *pdev)
 {
 	struct snd_soc_card *card = platform_get_drvdata(pdev);
-	struct machine_data *mdata =
-		container_of(card, struct machine_data, card);
+	struct machine_data *mdata = card_to_mdata(card);
 
 	snd_soc_unregister_card(card);
 	kfree(mdata);
-- 
2.55.0

